src/init.c

```diff
@@ -720,6 +768,367 @@ static int hyper_remove_container(char *json, int length)
 	return ret;
 }
 
+/**
+ * converts a line to an array, splits the line in tokens with strtok using
+ * horizontal spaces (\n\t ) as delimiter 
+ *
+ * \param line to process
+ * \param[out] array_size array size
+ * \param max_num_tokens max number of tokens that array should contain, if -1
+ * then there is no limit
+ *
+ * \return an array on success, else NULL
+ */
+static char **hyper_line_to_array(char *line, size_t *array_size, int max_num_tokens)
+{
+	char **array = NULL;
+	size_t size = 12;
+
+	if (max_num_tokens == 0) {
+		return NULL;
+	}
+
+	array = (char **)calloc(size, sizeof(*array));
+	if (array == NULL) {
+		return NULL;
+	}
+
+	size_t i = 0;
+	char *delim = " \n\t";
+	char *tok = strtok(line, delim);
+	while (tok != NULL) {
+		array[i++] = tok;
+
+		if (i >= size-1) {
+			char **tmp = NULL;
+			size += size;
+			tmp = (char **)realloc(array, size * sizeof(*array));
+			if (tmp == NULL) {
+				/* try to reallocate once again */
+				tmp = (char **)realloc(array, size * sizeof(*array));
+			}
+			if (tmp == NULL) {
+				/* realloc failed, free memory and return NULL */
+				free(array);
+				return NULL;
+			}
+			array = tmp;
+		}
+
+		if (max_num_tokens > 0 && (--max_num_tokens) == 0) {
+			break;
+		}
+
+		tok = strtok(NULL, delim);
+	}
+
+	if (array_size != NULL) {
+		*array_size = i;
+	}
+
+	/* Add NULL terminator */
+	array[i] = NULL;
+
+	return array;
+}
+
+/**
+ * gets the index of PID, for example in the next line pid index is 1
+ *    UID PID PPID C STIME TTY TIME CMD
+ *
+ * \param line to process
+ *
+ * \return pid index on success, else -1
+ */
+static int hyper_get_pid_index(const char *line)
+{
+	char **array = NULL;
+	char *new_line = NULL;
+	int ret = -1;
+	const char* pid_str = "PID";
+	const size_t pid_str_size = 3;
+
+	new_line = strdup(line);
+	if (new_line == NULL) {
+		fprintf(stderr, "can not duplicate string: %s\n", line);
+		return -1;
+	}
+
+	array = hyper_line_to_array(new_line, NULL, -1);
+	if (array == NULL) {
+		fprintf(stderr, "can not convert line to array: %s\n", new_line);
+		goto out;
+	}
+
+	for (size_t i=0; array[i] != NULL; i++) {
+		if (strncmp(array[i], pid_str, pid_str_size) == 0) {
+			ret = i;
+			break;
+		}
+	}
+
+out:
+	free_if_set(new_line);
+	free_if_set(array);
+	return ret;
+}
+
+/**
+ * appends current ps line to final ps output
+ *
+ * \param[in,out] ps_data \ref hyper_ps_data
+ *
+ * \return true on success, else false
+ */
+static bool hyper_append_ps_line(struct hyper_ps_data *ps_data)
+{
+	/* +1 for \n */
+	size_t line_len = ps_data->line_len + 1;
+	size_t output_len = *ps_data->output_len;
+	const size_t buf_size = 1024;
```

Comment:
This limit might be too low: on a test box, I've got a qemu cmdline that is `2634` bytes long.